Cgrates PANIC and Crash with LCR Supplier with empty supplier_ratingplan_ids

Hi,
Cgrates Crash if for a given supplier sorting is set to *least_cost and supplier_ratingplan_ids is empty
Here is test data for suppliers for testing

```
"pk","tpid","tenant","id","filter_ids","activation_interval","sorting","sorting_parameters","supplier_id","supplier_filter_ids","supplier_account_ids","supplier_ratingplan_ids","supplier_resource_ids","supplier_stat_ids","supplier_weight","supplier_blocker","supplier_parameters","weight","created_at"
"36","tpaql","cgrates.org","SPL_GROUP_1","FLT_SPL1",,"*least_cost",,"AQL",,,"RP_AQL_ON_PEAK",,,"20","0",,"20","25/1/2019 11:17:38"
"37","tpaql","cgrates.org","SPL_GROUP_1",,,"*least_cost",,"IDT",,,,,,"19","0",,"18","25/1/2019 11:35:10"
"38","tpaql","cgrates.org","SPL_GROUP_1",,,"*least_cost",,"AQL1",,,"RP_AQL_ON_PEAK",,,"19","0",,"18","25/1/2019 11:35:04"
```
Here is panic

> "RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> 2019/01/22 14:32:45 CGRateS <CGRAsterisk> [DEBUG] field <cgr_subsystems> as extra since is not present in ["Account","AnswerTime","CGRID","Category","Cost","CostSource","Destination","OriginHost","OriginID","Partial","PreRated","RequestType","RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> 2019/01/22 14:32:45 CGRateS <CGRAsterisk> [DEBUG] field <DisconnectCause> as extra since is not present in ["Account","AnswerTime","CGRID","Category","Cost","CostSource","Destination","OriginHost","OriginID","Partial","PreRated","RequestType","RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> 2019/01/22 14:32:45 CGRateS <CGRAsterisk> [DEBUG] field <Supplier> as extra since is not present in ["Account","AnswerTime","CGRID","Category","Cost","CostSource","Destination","OriginHost","OriginID","Partial","PreRated","RequestType","RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> 2019/01/22 14:32:45 CGRateS <CGRAsterisk> [ERROR] Destination 923345401254 not authorized for account: cgrates.org:1001, subject: *out:cgrates.org:call:1002
> 2019/01/22 14:34:05 CGRateS <CGRAsterisk> [DEBUG] field <EventName> as extra since is not present in ["Account","AnswerTime","CGRID","Category","Cost","CostSource","Destination","OriginHost","OriginID","Partial","PreRated","RequestType","RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> 2019/01/22 14:34:05 CGRateS <CGRAsterisk> [DEBUG] field <Supplier> as extra since is not present in ["Account","AnswerTime","CGRID","Category","Cost","CostSource","Destination","OriginHost","OriginID","Partial","PreRated","RequestType","RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> 2019/01/22 14:34:05 CGRateS <CGRAsterisk> [DEBUG] field <cgr_subsystems> as extra since is not present in ["Account","AnswerTime","CGRID","Category","Cost","CostSource","Destination","OriginHost","OriginID","Partial","PreRated","RequestType","RunID","SetupTime","Source","Subject","Tenant","ToR","Usage","rated"]
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xbf5870]
> 
> goroutine 622 [running]:
> github.com/cgrates/cgrates/engine.(*RateInterval).GetMaxCost(...)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/rateinterval.go:379
> github.com/cgrates/cgrates/engine.(*CallDescriptor).GetCost(0xc0005ae840, 0xc0005ae840, 0x1c61280, 0x9d29229e000)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/calldesc.go:593 +0x150
> github.com/cgrates/cgrates/engine.(*SupplierService).costForEvent(0xc000526380, 0xc00052feb8, 0x0, 0x0, 0x0, 0xc000285900, 0x1, 0x4, 0xc0003f9650, 0x0, ...)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/suppliers.go:244 +0x97b
> github.com/cgrates/cgrates/engine.(*SupplierService).populateSortingData(0xc000526380, 0xc00052feb8, 0xc000187130, 0xc000417350, 0x1, 0xc00036a118, 0x0, 0x1)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/suppliers.go:302 +0x220
> github.com/cgrates/cgrates/engine.(*LeastCostSorter).SortSuppliers(0xc000246160, 0xc0002c11c8, 0x3, 0xc0001b1100, 0x3, 0x4, 0xc00052feb8, 0xc000417350, 0x0, 0x0, ...)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/spls_leastcost.go:42 +0x157
> github.com/cgrates/cgrates/engine.SupplierSortDispatcher.SortSuppliers(0xc0002de3c0, 0xc0002c11c8, 0x3, 0xc0002c1210, 0xb, 0xc0001b1100, 0x3, 0x4, 0xc00052feb8, 0xc000417350, ...)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/libsuppliers.go:154 +0xda
> github.com/cgrates/cgrates/engine.(*SupplierService).sortedSuppliersForEvent(0xc000526380, 0xc00052fea0, 0xc00025aad8, 0x2, 0x2)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/suppliers.go:403 +0x178
> github.com/cgrates/cgrates/engine.(*SupplierService).V1GetSuppliers(0xc000526380, 0xc00052fea0, 0xc000285880, 0x7f1b26223000, 0x0)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/engine/suppliers.go:482 +0x27e
> github.com/cgrates/cgrates/apier/v1.(*SupplierSv1).GetSuppliers(0xc000214010, 0xc00052fea0, 0xc000285880, 0x0, 0x0)
>         /root/Projects/Proj1/src/github.com/cgrates/cgrates/src/github.com/cgrates/cgrates/apier/v1/suppliers.go:102 +0x42
> reflect.Value.call(0xc0004b8240, 0xc000214050, 0x13, 0x115b201, 0x4, 0xc00025af18, 0x3, 0x3, 0xc000370980, 0xfe0780, ...)
>         /usr/local/go/src/reflect/value.go:447 +0x454
> reflect.Value.Call(0xc0004b8240, 0xc000214050, 0x13, 0xc000264f18, 0x3, 0x3, 0xc000285800, 0xc000367b60, 0xc000412301)
>         /usr/local/go/src/reflect/value.go:308 +0xa4
> net/rpc.(*service).call(0xc0002840c0, 0xc000176000, 0xc0004c2058, 0xc0004c2060, 0xc0004cc480, 0xc0001a84e0, 0x10ae0c0, 0xc00052fea0, 0x16, 0x1080680, ...)
>         /usr/local/go/src/net/rpc/server.go:384 +0x14e
> created by net/rpc.(*Server).ServeCodec
>         /usr/local/go/src/net/rpc/server.go:481 +0x47e

--- a/engine/spls_highestcost.go
+++ b/engine/spls_highestcost.go
@@ -19,6 +19,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>
 package engine
 
 import (
+	"fmt"
+
 	"github.com/cgrates/cgrates/utils"
 )
 
@@ -39,6 +41,12 @@ func (hcs *HightCostSorter) SortSuppliers(prflID string, suppls []*Supplier,
 		Sorting:         hcs.sorting,
 		SortedSuppliers: make([]*SortedSupplier, 0)}
 	for _, s := range suppls {
+		if len(s.RatingPlanIDs) == 0 {
+			utils.Logger.Warning(
+				fmt.Sprintf("<%s> RatingPlanIDs is empty for suplier with ID: %s",
+					utils.SupplierS, s.ID))
+			return nil, utils.NewErrMandatoryIeMissing("RatingPlanIDs")
+		}
 		if srtSpl, pass, err := hcs.spS.populateSortingData(ev, s, extraOpts); err != nil {
 			return nil, err
 		} else if pass && srtSpl != nil {

--- a/engine/spls_leastcost.go
+++ b/engine/spls_leastcost.go
@@ -19,6 +19,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>
 package engine
 
 import (
+	"fmt"
+
 	"github.com/cgrates/cgrates/utils"
 )
 
@@ -39,6 +41,12 @@ func (lcs *LeastCostSorter) SortSuppliers(prflID string, suppls []*Supplier,
 		Sorting:         lcs.sorting,
 		SortedSuppliers: make([]*SortedSupplier, 0)}
 	for _, s := range suppls {
+		if len(s.RatingPlanIDs) == 0 {
+			utils.Logger.Warning(
+				fmt.Sprintf("<%s> RatingPlanIDs is empty for suplier with ID: %s",
+					utils.SupplierS, s.ID))
+			return nil, utils.NewErrMandatoryIeMissing("RatingPlanIDs")
+		}
 		if srtSpl, pass, err := lcs.spS.populateSortingData(ev, s, extraOpts); err != nil {
 			return nil, err
 		} else if pass && srtSpl != nil {

